src/IAssetAdapter.ts

```diff
@@ -11,30 +11,35 @@ import {
     Web3Client,
 } from './UsdcAssetAdapter';
 import {
-    HtlcDetails as EuroHtlcDetails,
+    HtlcDetails,
     OasisClient,
-} from './EuroAssetAdapter';
+} from './FiatAssetAdapter';
 
 export enum SwapAsset {
     NIM = 'NIM',
     BTC = 'BTC',
     USDC = 'USDC',
     USDC_MATIC = 'USDC_MATIC',
     EUR = 'EUR',
+    CRC = 'CRC',
 }
 
+export type FiatSwapAsset = SwapAsset.EUR | SwapAsset.CRC;
+
 export type Transaction<TAsset extends SwapAsset> =
     TAsset extends SwapAsset.NIM ? NimiqTransactionDetails
     : TAsset extends SwapAsset.BTC ? BitcoinTransactionDetails
     : TAsset extends SwapAsset.USDC | SwapAsset.USDC_MATIC ? UsdcTransactionDetails
-    : TAsset extends SwapAsset.EUR ? EuroHtlcDetails
+    : TAsset extends SwapAsset.EUR ? HtlcDetails
+    : TAsset extends SwapAsset.CRC ? HtlcDetails
```

Comment:
Is it possible to do this?

```ts
    : TAsset extends FiatSwapAsset ? HtlcDetails
```

Reply:
Yes, I did it in my local. Not yet published